src/containers/org/components/assign-permissions-to-data-set-modal/AssignPermissionsToDataSetModal.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * @flow
+ */
+
+import React, { useCallback, useState } from 'react';
+
+import styled from 'styled-components';
+import {
+  ModalFooter as LUKModalFooter,
+  ModalHeader as LUKModalHeader,
+} from 'lattice-ui-kit';
+import { useRequestState } from 'lattice-utils';
+import { useDispatch } from 'react-redux';
+import { RequestStates } from 'redux-reqseq';
+import type { Principal, UUID } from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import StepConfirm from './StepConfirm';
+import StepSelectDataSet from './StepSelectDataSet';
+import StepSelectPermissions from './StepSelectPermissions';
+import StepSelectProperties from './StepSelectProperties';
+
+import { ModalBody } from '../../../../components';
+import { ASSIGN_PERMISSIONS_TO_DATA_SET, assignPermissionsToDataSet } from '../../../../core/permissions/actions';
+import { resetRequestState } from '../../../../core/redux/actions';
+import { PERMISSIONS } from '../../../../core/redux/constants';
+import { SEARCH_DATA_SETS_TO_ASSIGN_PERMISSIONS, clearSearchState } from '../../../../core/search/actions';
+
+const ModalHeader = styled(LUKModalHeader)`
+  padding: 30px 0;
+`;
+
+const ModalFooter = styled(LUKModalFooter)`
+  padding: 30px 0;
+`;
+
+const AssignPermissionsToDataSetModal = ({
+  onClose,
+  organizationId,
+  principal,
+  step,
+  stepBack,
+  stepNext,
+} :{
+  onClose :() => void;
+  organizationId :UUID;
+  principal :Principal;
+  step :number;
+  stepBack :() => void;
+  stepNext :() => void;
+}) => {
+
+  const dispatch = useDispatch();
+
+  const [assignPermissionsToAllProperties, setAssignPermissionsToAllProperties] = useState(true);
+  const [targetDataSetId, setTargetDataSetId] = useState('');
+  const [targetDataSetTitle, setTargetDataSetTitle] = useState('');
+  const [targetPermissionOptions, setTargetPermissionOptions] = useState([]);
+
+  const assignPermissionsToDataSetRS :?RequestState = useRequestState([PERMISSIONS, ASSIGN_PERMISSIONS_TO_DATA_SET]);
+
+  const onCleanUp = useCallback(() => {
+    dispatch(clearSearchState(SEARCH_DATA_SETS_TO_ASSIGN_PERMISSIONS));
+    dispatch(resetRequestState([ASSIGN_PERMISSIONS_TO_DATA_SET]));
+  }, [dispatch]);
+
+  const onConfirm = () => {
+    dispatch(
+      assignPermissionsToDataSet({
+        principal,
+        dataSetId: targetDataSetId,
+        permissionTypes: targetPermissionOptions.map((option) => option.value),
+        withProperties: assignPermissionsToAllProperties,
+      })
+    );
+  };
+
+  if (step === 0) {
+    return (
+      <>
+        <ModalHeader onClickClose={onClose} textTitle="Assign Permissions To Data Set" />
+        <ModalBody onCleanUp={onCleanUp}>
```

Comment:
Judging from the actions dispatched inside onCleanup, it seems the intention is to clean up when this whole AssignPermissionsToDataSetComponent is unmounted, rather than on each step. Based on the way this is currently set up, I would expect onCleanUp to trigger on each step. 

Not every instance of ModalBody needs to clean up after itself and it feels like you're a little too eager to justify the abstraction of the clean up effect you've created earlier.

In situations where `StepSelectDataSet` does need to clean up after itself, I would still say relocating clean up to be outside of the component isn't necessary. Making abstractions for native hook behavior is dangerous as it obfuscates when and why you might need them. 

It didn't even really save you any lines of code either as the cleanup function is wrapped in a useCallback that would otherwise not be needed

Reply:
Yes, the intention is to clean up after the workflow is complete, not in between steps. Strangely, it seems to behave correctly as it was implemented, but that's definitely not what I intended. I've switched to `useEffect`.